src/plugin-segment/index.js

```diff
@@ -42,10 +42,51 @@ module.exports = function (context) {
           {
             tagName: 'script',
             innerHTML: `
-              !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e<analytics.methods.length;e++){var t=analytics.methods[e];analytics[t]=analytics.factory(t)}analytics.load=function(e,t){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+e+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=t};analytics.SNIPPET_VERSION="4.1.0";
-                analytics.load("${apiKey}");
-                analytics.page();
-                }}();
+              !function() {
+                /*
+                 * Necessary to allow the cookie name to be accessible as a variable
+                 * from function components that don't use hooks (such as analytics.js)
+                 */
+                window.segmentPermissionCookie = '${segmentPermissionCookie}';
+
+                window.loadSegmentFormScript = () => {
+                  const segmentScript = document.createElement('script');
+                  segmentScript.async = true;
+                  segmentScript.src = '/js/segmentForm.js';
+                  document.body.appendChild(segmentScript);
+
+                  const formScript = document.createElement('script');
+                  formScript.async = true;
+                  formScript.src = '/js/forms.js';
+                  document.body.appendChild(formScript);
+                };
+
+                /*
+                 * We make the loading function accessible so that when the user accepts
+                 * the cookie policy, segment is loaded without the need to refresh
+                 */
+                window.loadSegment = () => {
```

Comment:
We should not test for analytics here, it is already managed in row 79. If for some reason `window.analytics` is defined or hasn't been initialized then we'd never load this even though we need to.